common/src/main/java/org/apache/sedona/common/Functions.java

```diff
@@ -90,6 +93,57 @@ public static Geometry boundary(Geometry geometry) {
     return boundary;
   }
 
+  public static Geometry expand(Geometry geometry, double uniformDelta) {
+    return expand(geometry, uniformDelta, uniformDelta, uniformDelta);
+  }
+
+  public static Geometry expand(Geometry geometry, double deltaX, double deltaY) {
+    return expand(geometry, deltaX, deltaY, 0);
+  }
+
+  public static Geometry expand(Geometry geometry, double deltaX, double deltaY, double deltaZ) {
+    if (geometry == null || geometry.isEmpty()) {
+      return geometry;
+    }
+
+    Coordinate[] coordinates = geometry.getCoordinates();
+    double minX = Double.MAX_VALUE;
+    double maxX = -Double.MAX_VALUE;
+    double minY = Double.MAX_VALUE;
+    double maxY = -Double.MAX_VALUE;
+    double minZ = Double.MAX_VALUE;
+    double maxZ = -Double.MAX_VALUE;
+
+    for (int i = 0; i < coordinates.length; i++) {
+      minX = Math.min(minX, coordinates[i].x);
+      maxX = Math.max(maxX, coordinates[i].x);
+      minY = Math.min(minY, coordinates[i].y);
+      maxY = Math.max(maxY, coordinates[i].y);
+      minZ = Math.min(minZ, coordinates[i].z);
+      maxZ = Math.max(maxZ, coordinates[i].z);
```

Comment:
When a Geometry does not have Z value, JTS will give NaN (https://locationtech.github.io/jts/javadoc/org/locationtech/jts/geom/Coordinate.html#getZ--). Does the `Math` comparison still work in this case?

Reply:
Yeah, it just returns a NaN, and I am skipping the Z values when Geometry doesn't have it. I don't see it being a problem.